java-manta-client/src/main/java/com/joyent/manta/client/multipart/AbstractMultipartManager.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ */
+package com.joyent.manta.client.multipart;
+
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.exception.MantaMultipartException;
+import com.joyent.manta.http.entity.ExposedByteArrayEntity;
+import com.joyent.manta.http.entity.ExposedStringEntity;
+import com.joyent.manta.http.entity.MantaInputStreamEntity;
+import org.apache.commons.lang3.Validate;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.FileEntity;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+
+/**
+ * Base class providing generic methods useful for {@link MantaMultipartManager}
+ * implementations.
+ *
+ * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
+ * @since 3.0.0
+ *
+ * @param <UPLOAD> Manta multipart upload object used to manage MPU state
+ * @param <PART> Manta multipart upload part object used to manage MPU part state
+ */
+abstract class AbstractMultipartManager<UPLOAD extends MantaMultipartUpload, PART extends MantaMultipartUploadPart>
```

Comment:
I think this javadoc would be a good place to provide a Big Theory Statement for the generics based structure.